packages/components/src/components/multi-select/multi-select/multi-select-utils.spec.ts

```diff
@@ -0,0 +1,470 @@
+  value: string;
+  selected: boolean;
+  highlighted: boolean;
+  disabled: boolean;
+  hidden: boolean;
+  theme?: Theme;
+};
+
+type GenerateMultiSelectOptionsParams = {
+  amount: number;
+  selectedIndices?: number[];
+  highlightedIndex?: number;
+  disabledIndex?: number;
+  hiddenIndex?: number;
+};
+
+export const generateMultiSelectOptions = (
+  { amount, selectedIndices = [], highlightedIndex, disabledIndex, hiddenIndex }: GenerateMultiSelectOptionsParams = {
+    amount: 3,
+    selectedIndices: [],
+  }
+): MultiSelectOption[] => {
+  return Array.from(Array(amount)).map<Option>((_, idx) => ({
+    value: `Value ${idx}`,
+    textContent: `Option ${idx}`,
+    selected: selectedIndices.includes(idx),
+    highlighted: highlightedIndex === idx,
+    disabled: disabledIndex === idx,
+    hidden: hiddenIndex === idx,
+  })) as MultiSelectOption[];
+};
+
+describe('syncMultiSelectOptionProps', () => {
+  it('should update theme and force update for mismatched options', () => {
+    const options = generateMultiSelectOptions();
+    options[0].theme = 'light';
+    options[1].theme = 'dark';
+
+    syncMultiSelectOptionProps(options, 'dark');
+
+    options.forEach((option) => {
+      expect(option.theme).toBe('dark');
+    });
+  });
+});
+
+describe('initNativeSelect', () => {
+  it('should return native select with added attributes and add native select to host', () => {
+    const spy = jest.spyOn(setAttributesUtils, 'setAttributes');
+    const syncNativeSelectSpy = jest.spyOn(multiSelectUtils, 'syncNativeSelect');
+
+    const host = document.createElement('p-multi-select');
+    const name = 'options';
+    const disabled = true;
+    const required = false;
+
+    const nativeSelect = initNativeSelect(host, name, disabled, required);
+
+    expect(nativeSelect instanceof HTMLSelectElement).toBe(true);
+    expect(spy).toHaveBeenCalledWith(nativeSelect, {
+      multiple: 'true',
+      'aria-hidden': 'true',
+      tabindex: '-1',
+      slot: 'select',
+    });
+    expect(syncNativeSelectSpy).toHaveBeenCalledWith(nativeSelect, name, disabled, required);
+    expect(host.firstChild).toBe(nativeSelect);
+  });
+});
+
+describe('syncNativeSelect', () => {
+  it('should synchronize attributes of native select element', () => {
+    const setAttributeSpy = jest.spyOn(setAttributeUtils, 'setAttribute');
+
+    const nativeSelect = document.createElement('select');
+    const toggleAttributeSpy = jest.spyOn(nativeSelect, 'toggleAttribute');
+    const name = 'testSelect';
+    const disabled = true;
+    const required = false;
+
+    syncNativeSelect(nativeSelect, name, disabled, required);
+
+    expect(setAttributeSpy).toHaveBeenCalledWith(nativeSelect, 'name', name);
+    expect(toggleAttributeSpy).toHaveBeenCalledWith('disabled', true);
+    expect(toggleAttributeSpy).toHaveBeenCalledWith('required', false);
+  });
+
+  it('should remove "disabled" and "required" attributes when not disabled and not required', () => {
```

Comment:
the description does not fit the test? There is no check, if `disabled` and `required` are removed